Example/RealmResultsController-iOSTests/RealmQueueManagerTests.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  RealmQueueManagerTests.swift
+//  RealmResultsController
+//
+//  Created by Pol Quintana on 10/12/15.
+//  Copyright © 2015 Redbooth. All rights reserved.
+//
+
+import Foundation
+import Quick
+import Nimble
+
+@testable import RealmResultsController
+
+private let blocksTimeout: UInt32 = 1
+
+class RealmQueueManagerSpec: QuickSpec {
+    private var queueManager = RealmQueueManager()
+    private var thread1BlockHasBeenFired = false
+    private var thread2BlockHasBeenFired = false
+    private var timeoutHasBeenReached = false
+    
+    func fireThread1Block() {
+        let queue = dispatch_queue_create("THREAD 1", DISPATCH_QUEUE_SERIAL)
+        Threading.executeOnQueue(queue, sync: true) {
+            self.queueManager.addOperation {
+                self.thread1BlockHasBeenFired = true
+                sleep(2)
+            }
+        }
+    }
+    
+    func fireThread2Block() {
+        let queue = dispatch_queue_create("THREAD 2", DISPATCH_QUEUE_SERIAL)
+        Threading.executeOnQueue(queue, sync: true) {
+            self.queueManager.addOperation {
+                self.thread2BlockHasBeenFired = true
+                sleep(2)
+            }
+        }
+    }
+    
+    func oneBlockFired() -> Bool {
+        let a = self.thread1BlockHasBeenFired || self.thread2BlockHasBeenFired
+        return a && !(self.thread1BlockHasBeenFired && self.thread2BlockHasBeenFired)
```

Comment:
If this is true
````
    thread1BlockHasBeenFired | thread2BlockHasBeenFired | result
-----------------------------------------------------------------------------------------------
    0                        | 0                         | 0
    0                        | 1                         | 1
    1                        | 0                         | 1
    1                        | 1                         | 0
````

Thant means: `return self.thread1BlockHasBeenFired ^ self.thread2BlockHasBeenFired` should do the work without breaking any brain :P